packages/fe-fpm-writer/test/unit/headersection.test.ts

```diff
@@ -0,0 +1,131 @@
+import type { Editor } from 'mem-fs-editor';
+import { create } from 'mem-fs-editor';
+import { create as createStorage } from 'mem-fs';
+import { join } from 'path';
+import { generateCustomHeaderSection } from '../../src/section';
+import { RequestGroupId, type CustomHeaderSection, DesignTime } from '../../src/section/types';
+import type { Manifest } from '../../src/common/types';
+import { Placement } from '../../src/common/types';
+import * as manifestSections from './sample/section/webapp/manifest.json';
+
+const testDir = join(__dirname, 'sample/headersection');
+
+describe('CustomHeaderSection generateCustomHeaderSection', () => {
+    let fs: Editor;
+    const manifest = JSON.parse(JSON.stringify(manifestSections));
+    manifest['sap.ui5'].routing.targets.sample.options.settings.content.body.sections = {};
+    // Basic custom header section object
+    const customHeaderSection: CustomHeaderSection = {
+        target: 'sample',
+        name: 'NewCustomHeaderSection',
+        folder: 'extensions/custom',
+        title: 'New Custom Header Section',
+        subTitle: 'Custom Header section subtitle',
+        edit: {
+            name: 'NewCustomHeaderSectionEdit',
+            folder: 'extensions/custom',
```

Comment:
I think there is missing test where folders are different  `folder` vs `edit.folder`, which associated to comment https://github.com/SAP/open-ux-tools/pull/1142/files#r1285789765